### PR TITLE
Check for minimum iptables version

### LINKED ISSUE #976: Clean up firewall rules
Found in logs. It is possible to have firewall rules in an unexpected state, which we do not handle:
```
Chain OUTPUT (policy ACCEPT 0 packets, 0 bytes)
    pkts      bytes target     prot opt in     out     source               destination         
      12     5688 ACCEPT     tcp  --  *      *       0.0.0.0/0            168.63.129.16        owner UID match 0
      18      936 DROP       tcp  --  *      *       0.0.0.0/0            168.63.129.16       
       0        0 ACCEPT     tcp  --  *      *       0.0.0.0/0            168.63.129.16        owner UID match 0
       0        0 ACCEPT     tcp  --  *      *       0.0.0.0/0            168.63.129.16        ctstate INVALID,NEW
```
We need to determine a safe way of cleaning up rules before enabling drops.

### LINKED ISSUE #977: Repeated firewall logs (CentOS 6.5 HPC image + WALA 2.2.19)
**Repro Steps:**
1. Deploy CentOS 6.5 HPC VM.
2. Upgrade WALA to latest using master branch.
3. Restart.

**Issue -**
1. waagent keeps dumping firewall related messages to  /var/log/waagent.log infinitely.
2. I've seen waagent.log file grow beyond 4GB.
3. If we allow this state, OS will run out of disk space.

### PR DESCRIPTION
**Pull request by boumenot**
Only enable the firewall if iptables is at least 1.4.11 or greater.

Closes #976
Closes #977

### CODE REVIEW

**Comment by boumenot** [comment]
This PR is not sufficient.  The iptables on CentOS 6.9 is 1.4.7, but it does support the -C option.  It appears that they backported this work, so a code change that relies solely on the version number is insufficient.

File: azurelinuxagent/common/exception.py [hunk 1]
```diff
@@ -44,7 +44,7 @@ def __init__(self, msg=None, inner=None):
 
 class AgentNetworkError(AgentError):
     """
-    When network is not avaiable.
+    When network is not available\.
```
Comment (hglkrijger) [inline azurelinuxagent/common/exception.py]: `\` at the end

File: azurelinuxagent/agent.py [hunk 2]
```diff
@@ -63,6 +63,9 @@ def __init__(self, verbose, conf_file_path=None):
                                  path="/var/log/waagent.log")
         logger.add_logger_appender(logger.AppenderType.CONSOLE, level,
                                  path="/dev/console")
+        logger.add_logger_appender(logger.AppenderType.TELEMETRY,
```
Comment (hglkrijger) [inline azurelinuxagent/agent.py]: Do you need to merge/rebase?

File: azurelinuxagent/common/osutil/default.py [hunk 3]
```diff
@@ -129,24 +133,40 @@ def get_firewall_will_wait(self):
                 else ""
         return wait
 
-    def remove_firewall(self):
+    def _delete_rule(self, rule):
+        """
+        Continually execute the delete operation until the return
+        code is non-zero.
+        """
+        while True:
```
Comment (hglkrijger) [inline azurelinuxagent/common/osutil/default.py]: `while True` makes me nervous; can we put in another escape hatch?

File: azurelinuxagent/ga/update.py [hunk 4]
```diff
@@ -284,10 +284,7 @@ def run(self):
                 utc_start = datetime.utcnow()
 
                 last_etag = exthandlers_handler.last_etag
-                try:
-                    exthandlers_handler.run()
-                except RestartError:
```
Comment (hglkrijger) [inline azurelinuxagent/ga/update.py]: Why remove this? Or do you need to sync?

File: azurelinuxagent/ga/update.py [hunk 5]
```diff
@@ -299,7 +296,7 @@ def run(self):
                         duration=elapsed_milliseconds(utc_start),
                         message="Incarnation {0}".format(
                                             exthandlers_handler.last_etag),
-                        log_event=exthandlers_handler.log_process)
+                        log_event=True)
```
Comment (hglkrijger) [inline azurelinuxagent/ga/update.py]: Why remove this?

File: azurelinuxagent/ga/update.py (hunk 4 above)
Comment (boumenot) [reply azurelinuxagent/ga/update.py]: I need to sync.

File: azurelinuxagent/ga/update.py (hunk 5 above)
Comment (boumenot) [reply azurelinuxagent/ga/update.py]: I need to sync.

File: azurelinuxagent/common/osutil/default.py (hunk 3 above)
Comment (boumenot) [reply azurelinuxagent/common/osutil/default.py]: Give me some suggestions.  What do you typically do?

File: azurelinuxagent/common/exception.py (hunk 1 above)
Comment (boumenot) [reply azurelinuxagent/common/exception.py]: I need to sync.

File: azurelinuxagent/agent.py (hunk 2 above)
Comment (boumenot) [reply azurelinuxagent/agent.py]: Yes.

File: azurelinuxagent/common/osutil/default.py (hunk 3 above)
Comment (hglkrijger) [reply azurelinuxagent/common/osutil/default.py]: Count or time, so keep removing rules until done or we hit some large number, for example.

**Pull request merged**